datadog/resource_datadog_security_monitoring_filters.go

```diff
@@ -0,0 +1,253 @@
+
+	response, httpResponse, err := datadogClientV2.SecurityMonitoringApi.CreateSecurityFilter(authV2, *filterCreate)
+	if err != nil {
+		return utils.TranslateClientErrorDiag(err, httpResponse, "error creating security monitoring filter")
+	}
+
+	// store the resource id
+	data := response.GetData()
+	d.SetId(data.GetId())
+
+	return nil
+}
+
+func resourceDatadogSecurityMonitoringFilterRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	providerConf := meta.(*ProviderConfiguration)
+	datadogClientV2 := providerConf.DatadogClientV2
+	authV2 := providerConf.AuthV2
+
+	id := d.Id()
+	filterResponse, httpResponse, err := datadogClientV2.SecurityMonitoringApi.GetSecurityFilter(authV2, id)
+	if err != nil {
+		if httpResponse != nil && httpResponse.StatusCode == 404 {
+			d.SetId("")
+			return nil
+		}
+		return diag.FromErr(err)
+	}
+
+	updateResourceDataFilterFromResponse(d, filterResponse)
+
+	// handle warning
+	if filterResponse.HasMeta() {
+		filterMeta := filterResponse.GetMeta()
+		warning := filterMeta.GetWarning()
+		diagnostic := diag.Diagnostic{Severity: diag.Warning, Summary: warning}
+		return diag.Diagnostics{diagnostic}
+	}
+	return nil
+}
+
+func resourceDatadogSecurityMonitoringFilterUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	providerConf := meta.(*ProviderConfiguration)
+	datadogClientV2 := providerConf.DatadogClientV2
+	authV2 := providerConf.AuthV2
+
+	filterId := d.Id()
+
+	response, diagnostics, failed := fetchFilterFromApi(datadogClientV2, authV2, filterId)
+	if failed {
+		return diagnostics
+	}
+
+	filterUpdate := buildSecMonFilterUpdatePayload(response, d)
+
+	if _, httpResponse, err := datadogClientV2.SecurityMonitoringApi.UpdateSecurityFilter(authV2, filterId, *filterUpdate); err != nil {
+		return utils.TranslateClientErrorDiag(err, httpResponse, "error updating security monitoring filter")
+	}
+
+	return nil
+}
+
+func fetchFilterFromApi(datadogClientV2 *datadogV2.APIClient, authV2 context.Context, filterId string) (datadogV2.SecurityFilterResponse, diag.Diagnostics, bool) {
+	response, httpResponse, err := datadogClientV2.SecurityMonitoringApi.GetSecurityFilter(authV2, filterId)
+
+	if err != nil {
+		if httpResponse != nil && httpResponse.StatusCode == 404 {
+			return datadogV2.SecurityFilterResponse{}, diag.FromErr(errors.New("default rule does not exist")), true
+		}
+
+		return datadogV2.SecurityFilterResponse{}, utils.TranslateClientErrorDiag(err, httpResponse, "error fetching filter"), true
+	}
+	return response, nil, false
+}
+
+func resourceDatadogSecurityMonitoringFilterDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	providerConf := meta.(*ProviderConfiguration)
+	datadogClientV2 := providerConf.DatadogClientV2
+	authV2 := providerConf.AuthV2
+
+	filterId := d.Id()
+
+	_, diagnostics, failed := fetchFilterFromApi(datadogClientV2, authV2, filterId)
```

Comment:
Same here.

Reply:
here, it's indeed useless, I'll remove it.